COBE-11 Configurar Spring Security

josed1608 opened the pull request:
Se configuró Spring Security para generar tokens JWT con los permisos, el rol y el nombre de usuario en el payload. Se creó un endpoint auth/signin para probar el signin y que devuelva el token. Ahorita el passwordEncoder no hace nada, entonces las contraseñas en la base de datos pueden estar en plain text para probar

josed1608 committed:
COBE-11 Configurar Spring Security
Se configuró Spring Security para generar tokens JWT con los permisos, el rol y el nombre de usuario en el payload

PauloBarrantes commented on src/main/java/com/coopetico/coopeticobackend/controladores/AuthControlador.java:
```diff
@@ -0,0 +1,56 @@
+
+@RestController
+@RequestMapping("/auth")
+public class AuthControlador {
```
Hace falta la documentación en Javadoc

PauloBarrantes reviewed (changes requested):
Por favor agregar la documentación siguiendo el patrón de Javadoc, pues es algo que van a usar todos y que quede claro todo.

josed1608 committed:
COBE-11 Configurar Spring Security
Se agregaron comentarios Javadoc y además se arreglaron detalles del JWT.

josed1608 committed:
COBE-11 Configurar Spring Security
Se aflojó la seguridad para los endpoints a los que no se les especifique restricción

PauloBarrantes reviewed (approved):
Linda la doc
